merino-cache/src/redis/mod.rs

```diff
@@ -125,6 +130,99 @@ where
         }
     }
 
+    /// Generate a lock identifier key.
+    /// This is a VERY simple locking mechanism. The only bit of fancy is that it will
+    /// expire, allowing for "stuck" queries to eventually resolve.
+    fn pending_key(key: &str) -> String {
+        format!("pending_{}", key)
+    }
+
+    /// See iff a record update is pending. (Does not check lock value, only if it exists)
+    async fn is_pending(&self, key: &str) -> Result<bool, SuggestError> {
+        let pending_key = Self::pending_key(key);
+        let mut connection = self.redis_connection.clone();
+
+        tracing::trace!("🔒Checking key: {:?}", &pending_key);
```

Comment:
```suggestion
        tracing::trace!(%pending_key, "🔒Checking key");
```

You can do string interpolation, but structured logging in Tracing is done separate from that.